syntax = "proto2";

package report;

enum Alignment {
    ALLY = 0;
    ENEMY = 1;
}

enum Class {
        CORVETTE = 0;
        FRIGATE = 1;
        CRUISER = 2;
        DESTROYER = 3;
        CARRIER = 4;
        DREADNOUGHT = 5;
}

message Officer {
    optional string first_name = 1;
    optional string last_name = 2;
    optional string rank = 3;
}

message Coordinate {
    repeated double values = 1;
}

message Spaceship {
    optional Alignment alignment = 1;

    optional string name = 2;
    optional Class type = 3;
    optional float length = 4;
    optional int32 size = 5;
    optional bool armed = 6;

    repeated Officer officers = 7;
}

service ReportingService {
    rpc GetShips(Coordinate) returns (stream Spaceship) {}
}


